src/main/src/main/java/org/geoserver/config/CatalogModificationUserUpdater.java

```diff
@@ -0,0 +1,90 @@
+/* (c) 2019 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.config;
+
+import java.util.logging.Logger;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.CatalogException;
+import org.geoserver.catalog.CatalogInfo;
+import org.geoserver.catalog.PublishedInfo;
+import org.geoserver.catalog.StoreInfo;
+import org.geoserver.catalog.StyleInfo;
+import org.geoserver.catalog.WorkspaceInfo;
+import org.geoserver.catalog.event.CatalogAddEvent;
+import org.geoserver.catalog.event.CatalogBeforeAddEvent;
+import org.geoserver.catalog.event.CatalogListener;
+import org.geoserver.catalog.event.CatalogModifyEvent;
+import org.geoserver.catalog.event.CatalogPostModifyEvent;
+import org.geoserver.catalog.event.CatalogRemoveEvent;
+import org.geotools.util.logging.Logging;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+public class CatalogModificationUserUpdater implements CatalogListener {
+
+    static final Logger LOGGER = Logging.getLogger(CatalogModificationUserUpdater.class);
+
+    public static final String TRACK_USER = "TRACK_USER";
+
+    Catalog catalog;
+
+    GeoServer geoServer;
+
+    public CatalogModificationUserUpdater(Catalog catalog, GeoServer geoServer) {
+        this.catalog = catalog;
+        this.geoServer = geoServer;
+        catalog.addListener(this);
+
+        LOGGER.fine("Initiated CatalogModificationUserUpdater");
+    }
+
+    @Override
+    public void handlePreAddEvent(CatalogBeforeAddEvent event) throws CatalogException {
+        SettingsInfo settings = geoServer.getSettings();
+        String property = System.getProperty(TRACK_USER, "false");
```

Comment:
Kubernetes deploys normally prefer the usage of environment variables.
Rather than restricting the user to a given approach, use ``GeoServerExtensions.getProperty``, it will check system, environment, and servlet context. 

It's also handy for testing, you can use GeoServerExtensionsHelper.property to set a property value for the test, and GeoServerExtensionsHelper.clear to clear it afterwards.